File: 2023-11-10/a/main.py


if __name__ == '__main__':
    cases = int( input() )
    while cases:
        cases -= 1

        book  = list(map(float, input().split(' ')))
        candles = int( input() )
        light = False
        while candles:
            candles -= 1
            candle = list(map(float,input().split(' ')))

            dist = (
                (book[0] - candle[0])**2 
                + (book[1] - candle[1])**2
            )**0.5
            print(dist)    
            if dist < 8.0:
                light = True
                break
        if light:
            print('light a candle')
        else:
            print('curse the darkness')

File: 2023-11-10/a/main.cpp
#include <iostream>
#include <cmath>


int main(){
  int N;
  std::cin >> N;
  while(N){
    N--; 
    double bx, by;
    std::cin >> bx >> by;

    int C;
    std::cin >> C;

    bool light = false;

    while(C){
      C--; 
      double cx, cy;
      std::cin >> cx >> cy;

      double dist;
      dist = sqrt(pow((bx-cx),2) + pow((by-cy),2));
      if(dist < 8.0)
        light = true;
    }

    if(light){
      std::cout << "light a candle" << std::endl;
    }
    else{
      std::cout << "curse the darkness"  << std::endl;
    }

  }
  return 0;
}

